app/code/Magento/Catalog/Model/Product/Option/Type/Select.php

```diff
@@ -30,23 +30,35 @@ class Select extends \Magento\Catalog\Model\Product\Option\Type\DefaultType
      */
     protected $string;
 
+    /**
+     * @var array
+     */
+    private $singleSelectionTypes;
+
     /**
      * @param \Magento\Checkout\Model\Session $checkoutSession
      * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
      * @param \Magento\Framework\Stdlib\StringUtils $string
      * @param \Magento\Framework\Escaper $escaper
      * @param array $data
+     * @param array $singleSelectionTypes
      */
     public function __construct(
         \Magento\Checkout\Model\Session $checkoutSession,
         \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
         \Magento\Framework\Stdlib\StringUtils $string,
         \Magento\Framework\Escaper $escaper,
-        array $data = []
+        array $data = [],
+        array $singleSelectionTypes = []
     ) {
         $this->string = $string;
         $this->_escaper = $escaper;
         parent::__construct($checkoutSession, $scopeConfig, $data);
+
+        $this->singleSelectionTypes = $singleSelectionTypes ?: [
+            \Magento\Catalog\Api\Data\ProductCustomOptionInterface::OPTION_TYPE_DROP_DOWN,
+            \Magento\Catalog\Api\Data\ProductCustomOptionInterface::OPTION_TYPE_RADIO,
+        ];
```

Comment:
@ihor-sviziev, could you please clarify, why we need default values here, cause we already have them in di.xml?
Also, here we have numeric array, and in di.xml we have associative array for default values.

Reply:
Hi @nmalevanec,
It's needed for backward compatiblity, when last argument will not be sent. For instance when someone extended this class in his code when this attribute was missing.

Reply:
Hi @ihor-sviziev , consider defining default arguments in the di.xml, with this approach this parameter will become configurable instead of hardcoded.

Reply:
Hi @sidolov,
As we discussed in Slack - it's needed for backward compatibility, in case if someone will extend this class & will not pass last argument in `parent::__construct(...)` - we have to have default value, similar as we're getting object from object manager by fallback.